### PR TITLE
📝 docs: INSTALL.md

### PR DESCRIPTION
**Pull request by peb-peb**
## Description

The above document would provide a detailed description to new contributors to setup `gcp_scanner` on their local environment. 

## Changes Made
- added `INSTALL.md` docs for the installation guide on the tool.

## Checklist
- [ ] I have read and followed the contributing guidelines.
- [x] I have tested my changes thoroughly and they work as expected.
- [x] I have added necessary tests for the changes made.
- [ ] I have updated the documentation to reflect the changes made.
- [x] My code follows the project's coding style and standards.
- [x] I have added appropriate commit messages and comments for my changes.

## Additional Notes
- should we add any other points or links in this?
  - like the pypi 
- I have added each code in triple ticks ``` (or codeblock) instead of a single tick ` as this enables the **copy** button.

### CODE REVIEW

**Comment by peb-peb** [comment]
@mshudrak I do not have permission to create a `wiki`. So, I have added it here. Once the wiki is done, we can either keep this or remove it.

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# 📝 Installation for GCP Scanner
+
+## Install basic dependencies
+
+**Linux** is the preferred operating system to use while contributing to GCP Scanner. If you're using Windows, we recommend setting up [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
```
Comment (mshudrak) [inline INSTALL.md:5]: Could you open a bug to add the same description but for MacOS?

**Review by mshudrak — CHANGES_REQUESTED** [review]
overall looks good, thanks for commit. PTAL at my suggestions

File: INSTALL.md (hunk 1 above)
Comment (peb-peb) [reply INSTALL.md:5]: Done :+1: 
See #125

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    ```bash
+    pylint --rcfile pylintrc --disable=W0703,R1734,R1735,C0209,C0103,R1732 src/gcp_scanner/*.py
+    ```
+
+2. Ensure that the corresponding tests are successful.
+
+3. If any new features have been added, then check with GCP to ensure they work as expected.
+
+### Building a standalone binary with PyInstaller
```
[the remark this answers is not in the document]
Comment (peb-peb) [reply INSTALL.md:111]: ok :+1:

File: INSTALL.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+**Linux** is the preferred operating system to use while contributing to GCP Scanner. If you're using Windows, we recommend setting up [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
+
+- Have `python` installed on your system. Check out this [link](https://www.python.org/downloads/) to download.
+- have `git` installed on your system.
+
+## Steps to follow to setup gcp-scanner
+
+### To install the package
+
+Install the package using `pip`
```
[the remark this answers is not in the document]
Comment (peb-peb) [reply INSTALL.md:14]: this is using `pypi`? If yes, then I'm not able to find it with a quick google search.

File: INSTALL.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    pip install .
+    ```
+
+7. Run to check if the tool was installed correctly
+
+    ```bash
+    gcp-scanner --help
+    ```
+
+### Some other things to keep in mind before any PR
```
[the remark this answers is not in the document]
Comment (peb-peb) [reply INSTALL.md]: Done :+1: 

Also, initially, I thought of putting all these to CONTRIBUTING.md itself and maybe segregating them if they become too large. Should we move with that approach instead?

Advantages of the current approach -
- makes the documenting process much more modular and clean.

Disadvantages of the current approach - 
- New contributors may find it hard to navigate.

File: INSTALL.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mshudrak) [reply INSTALL.md:14]: You can just test if it works for you using `pip` :)

**Review by mshudrak — APPROVED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -60,6 +64,24 @@ This project follows [Google's Open Source Community Guidelines](https://opensou
 
 **Example:** docs/11/update-installation-instructions
  
+### Some other things to keep in mind before any PR
+
+1. Check for linting using PyLint:
+    - for first time do: (to install a local copy of `pylintrc`)
```
Comment (mshudrak) [inline CONTRIBUTING.md:70]: Small nit `:` should be after  `(to install a local copy of `pylintrc`)`

File: INSTALL.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+### To install the package
+
+Install the package using `pip`
+
+```bash
+pip install gcp-scanner
+```
+
+Check if installation was done correctly -
```
Comment (mshudrak) [inline INSTALL.md:20]: Nit remove, trailing `-`

Comment (mshudrak) [reply INSTALL.md:20]: in other places as well

File: INSTALL.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (peb-peb) [reply INSTALL.md:14]: It works. Yay!

File: INSTALL.md (hunk 6 above)
Comment (peb-peb) [reply INSTALL.md]: done :+1:

**Pull request merged**